src/fairseq2/models/asr/_model.py

```diff
@@ -22,7 +24,7 @@ class AsrModel(Module, ABC):
     """Represents an Automatic Speech Recognition model."""
 
     @abstractmethod
-    def forward(self, batch: SequenceBatch) -> AsrModelOutput: ...
+    def forward(self, batch: SequenceBatch | Seq2SeqBatch) -> AsrModelOutput: ...
```

Comment:
let's see where it brings us, but otherwise let not spend a lot of time on it

Reply:
Model definition was refactored (https://github.com/facebookresearch/fairseq2/blob/v0.5.0a1/src/fairseq2/models/wav2vec2/asr/model.py), does not need to be included in the up-level.